src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java

```diff
@@ -196,4 +201,20 @@ public AlwaysStartOnPrimaryConnectionStrategy maxPauseSec(long maxPauseSec) {
         this.maxPauseSec = maxPauseSec;
         return this;
     }
+
+    @Override
+    public @Nullable InetSocketAddress localSocketBinding() {
+        // Create a new InetSocketAddress each time to allow host to be re-resolved
+        return localSocketBindingHost != null ? new InetSocketAddress(localSocketBindingHost, localSocketBindingPort) : null;
```

Comment:
It's good to create a new InetSocketAddress each time because it will trigger re-resolution if the host is configured as a domain name. InetSocketAddresses, once resolved will never re-resolve.